### ISSUE TITLE
Question about training objectives

### ISSUE THREAD

**Issue by jwh97nn** [baaivision/Emu#16]
In Section 2.2 of the paper, it is mentioned that there are two types of losses:
- ordinary language modeling cross-entropy loss for discrete text tokens.
- L2 regression loss for continuous visual embeddings.

And I'm wondering how the L2 loss is calculated.
Based on my understanding, it's the difference between **the predicted token** (after the regression head) and **that token outputed from the Causal Transformer**.
Can you confirm if my understanding is correct or provide more information on how the L2 regression loss is calculated?
Thank you

**Comment by yqy2001** [comment]
Thank you for your question. The loss is computed between the predicted embeddings (after the regression head) and the embeddings fed into the Multimodal LLM.

Note that the embedding dim outputted from Causal Transformer is 768, which is transformed into 5120 (the dim of LLaMA-13B) with a linear layer. We compute regression loss with dim 5120.

**Comment by yqy2001** [comment]
Thank you for your question. The loss is computed between the predicted embeddings (after the regression head) and the embeddings fed into the Multimodal LLM.

Note that the embedding dim outputted from Causal Transformer is 768, which is transformed into 5120 (the dim of LLaMA-13B) with a linear layer. We compute the regression loss with dim 5120.

**Comment by jwh97nn** [comment]
Thank you for your explanation.
